encoding.go

```diff
@@ -10,15 +11,20 @@ import (
 	"github.com/emersion/go-textwrapper"
 )
 
-type unknownEncodingError struct {
-	error
+type EncodingError struct {
```

Comment:
I suggested earlier to rename `UnknownEncodingError` to `EncodingError`. After thinking about this: encoding-related errors can happen when creating the encoding decoder, and when the data being decoded is malformed. We only use this error for the first kind, so maybe in the end it makes more sense to keep the name `UnknownEncodingError`. What do you think?

At any rate, naming should be consistent between encoding and charset errors: right now we have `EncodingError` and `UnknownCharsetError`, we need to at least change either to match the other.

Sorry for the change of mind.

Reply:
Fine for me, I don't mind either way so let's just keep the naming as it was.

I'll send an update later